### DISCUSSION TITLE
Selective static site regeneration

### DISCUSSION THREAD

**Discussion by qwertyforce** [vercel/next.js#19819] [Help]
When i do npm run build, it generates all the paths.
Is there a way to regenerate a certain path?
For example, I want to statically regenerate /image/[id] and I don't want to regenerate /news/[id]

**Comment by leerob** [comment]
https://nextjs.org/docs/basic-features/data-fetching#incremental-static-regeneration

**Reply by qwertyforce** [reply]
If i understand correctly, this will regenerate a certain page after a request, but I want to regenerate specific path (/path/1, /path/2, /path/N) manually

**Reply by leerob** [reply]
Correct. It's not possible to regenerate an entire path manually. You could, however, just put the same revalidation period on all of the files. Then it works similarly. You can't currently programmatically clear the cache.
